relayer/app/worker.go

```diff
@@ -191,6 +191,12 @@ func (w *Worker) newCallback(
 	msgStreamMapper msgStreamMapper,
 ) cchain.ProviderCallback {
 	return func(ctx context.Context, att xchain.Attestation) error {
+		if ok, err := att.Verify(); !ok {
```

Comment:
I had a look at the cprovider where it receives the data over the wire in `cchain/provider/abc.go`. We already do this in `atypes.AttestationFromProto` which is called as soon as we get the data over the wire. I think that is actually sufficient. As that is the point where we get the data over the wire. So I don't think we need to perform duplicate verification.

Reply:
We should probably add "duplicate" signature verification to `halo/attest/types/tx.go::Attestation.Verify`

Reply:
https://github.com/omni-network/omni/pull/2137/commits/78b35136c5ff9dc7b8c2efcdbd6a8a24f3b87431